src/plugins/kibana/public/settings/sections/advanced/advanced_row.html

```diff
@@ -66,9 +75,10 @@
 
     <!-- Setting display formats -->
     <span ng-if="!conf.editing" data-test-subj="currentValue">
-      <span ng-show="(conf.normal || conf.json || conf.select)">{{conf.value || conf.defVal}}</span>
-      <span ng-show="conf.array">{{(conf.value || conf.defVal).join(', ')}}</span>
-      <span ng-show="conf.bool">{{conf.value === undefined ? conf.defVal : conf.value}}</span>
+      <span ng-if="(conf.normal || conf.json || conf.select)">{{conf.value || conf.defVal}}</span>
+      <span ng-if="conf.array">{{(conf.value || conf.defVal).join(', ')}}</span>
+      <span ng-if="conf.bool">{{conf.value === undefined ? conf.defVal : conf.value}}</span>
+      <span ng-if="conf.markdown" ng-bind-html="conf.value | markdown"></span>
```

Comment:
Since you use the markdown filter here, you should also import `public/filters/markdown` in `src/plugins/kibana/public/settings/sections/advanced/advanced_row.js`